### ISSUE TITLE
Checkbox requires two clicks (Concurrent Mode)

### ISSUE THREAD

**Issue by MarcusCemes** [chakra-ui/chakra-ui#2350]
# Bug report

## Describe the bug

The `<Checkbox />` component is missing some accessibility features when used in a React project with Concurrent Mode.

- The element must be focussed with an initial click before it can be toggled with a second click
- The spacebar has no effect on toggling the focussed component
- When the spacebar is pressed, an additional click is required to toggle the component

## To reproduce

1. Create a new React project
2. Enable the unstable Concurrent Mode
3. Render and interact with a `<Checkbox />` component

## Minimal reproduction

https://codesandbox.io/s/chakraui-checkbox-bug-concurrent-mode-d9rm9?file=/src/index.js

The index file has a constant `ENABLE_CONCURRENT` that can be easily toggled to enable/disable the bug. The issue is present when concurrent mode is enabled. Switching React versions is not necessary, and the component behaves normally when concurrent mode is disabled, even with the experimental React version.

## Expected behavior

The checkbox should have the same behaviour as when legacy React is used.

## Screenshots

Not applicable.

## System information

- OS: Windows
- Browser (if applies): Chrome
- Version of @chakra-ui/core: 1.0.0-rc.7
- Version of React: 0.0.0-experimental-4ead6b530

## Additional context

Let me be clear, this issue is present on the experimental version of React _and_ the unreleased preview of Chakra UI. I'm not surprised that it is broken, but wanted to give a heads up for the future release of React! I'm not sure whether you want to track such issues.

I will keep on using the experimental version of React and can keep this issue updated in case it gets resolved by the next experimental release.

**Comment by ljosberinn** [comment]
Thanks for filing this so soon - we haven't checked whether Chakra is CM compatible at all as its still to early to tell as the React team has explained recently. We'll explore this once CM is in alpha but frankly said, I'm confident Chakra is not fully compatible with CM due to the implicated required changes (pure renders e.g. are an issue for framer-motion).

Please permit me to close this until CM is in alpha.
